File: finagle-example/src/main/scala/com/twitter/finagle/example/thrift/ThriftServer.scala
package com.twitter.finagle.example.thrift

import com.twitter.finagle.example.thriftscala.Hello
import com.twitter.finagle.Thrift
import com.twitter.util.{Await, Future}

object ThriftServer {
  def main(args: Array[String]): Unit = {
    //#thriftserverapi
    val server = Thrift.server.serveIface(
      "localhost:8080",
      new Hello.MethodPerEndpoint {
        def hi() = Future.value("hi")
      })
    Await.ready(server)
    //#thriftserverapi
  }
}

File: finagle-example/src/main/java/com/twitter/finagle/example/java/thrift/ThriftServer.java
package com.twitter.finagle.example.java.thrift;

import com.twitter.finagle.ListeningServer;
import com.twitter.finagle.Thrift;
import com.twitter.finagle.example.thriftscala.Hello;
import com.twitter.util.Await;
import com.twitter.util.Closable;
import com.twitter.util.Closables;
import com.twitter.util.Future;
import com.twitter.util.TimeoutException;

public final class ThriftServer {

  private ThriftServer() { }

  public static class HelloImpl implements Hello.MethodPerEndpoint {
    public Future<String> hi() {
      return Future.value("hi");
    }

    @Override
    public Closable asClosable() {
      return Closables.NOP;
    }
  }

  /**
   * Runs the example with given {@code args}.
   *
   * @param args the argument list
   */
  public static void main(String[] args) throws TimeoutException, InterruptedException {
    //#thriftserverapi
    Hello.MethodPerEndpoint impl = new HelloImpl();
    ListeningServer server = Thrift.server().serveIface("localhost:8080", impl);
    Await.ready(server);
    //#thriftserverapi
  }
}
